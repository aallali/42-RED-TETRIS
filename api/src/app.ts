
import * as express from "express";
import * as cors from "cors"
import { Socket } from "socket.io";
import Game from "./Game"

import ROOM from "./Room"
import PLAYER from "./Player"
import { ROOM_MODE } from "./types"


const log = console.log
interface IPlayer {
	id: string;
	nickname: string;
}
interface IRoom {
	id: string;
	name: string;
	players: []
	admin: IPlayer
}
class App {

	public app: express.Application;
	public http: any
	public io: Socket
	private port: number = 4242
	private gameIns: Game
	constructor() {
		this.app = express();
		this.app.use(
			cors({
				origin: "*",
				credentials: true,
			})
		);
		this.http = require("http").Server(this.app);
		this.io = require("socket.io")(this.http, {
			pingInterval: 60000,
			cors: {
				origin: "*",
			},
		});
		this.routes()
		this.initSocketEvents()
		this.gameIns = new Game(this.io)

	}
	/**
	 * 
	 */
	private routes() {
		this.app.get("/", (req: any, res: any) => {
			console.log(Array.from(this.gameIns.players))
			res.send(Array.from(this.gameIns.players))
		});
	}
	private initSocketEvents() {
	}
	public listen() {
		console.clear()
		this.http.listen(this.port, () =>
			console.log(`ðŸš€ Server is running on PORT ${this.port}`)
		);
	}
}

export default new App();
