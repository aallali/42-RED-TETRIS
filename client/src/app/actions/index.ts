import {
	SET_PLAYER, SET_GAME,
	LOGOUT_PLAYER, UPDATE_SCORE,
	SET_PLAYER_ADMIN, ADD_ROW,
	PLAYER_LOST, RESET_STATES,
	BROADCAST_SCORE,
	UNSET_PLAYER_ADMIN
} from "../../reducers/player.reducer";
import { UPDATE_PLAYERS, CLEAR_OPPONENTS } from "../../reducers/opponent.reducer"
import { SET_ERROR, CLEAR_ERROR } from "../../reducers/error.reducer"

import {
	SHIFT_TETRO, MORE_TETROS,
	SET_GAME_MODE, SET_GAME_OVER,
	SET_GAME_STARTED, SET_IN_GAME,
	SET_WINNER, RESET_GAME,
	UPDATE_GAME_MODE,
	LEAVE_GAME,
	SET_GAME_TITLE
} from "../../reducers/game.reducer"

export {
	// Players actions
	SET_PLAYER, SET_GAME,
	LOGOUT_PLAYER,
	UPDATE_SCORE,
	SET_PLAYER_ADMIN,
	ADD_ROW, PLAYER_LOST,
	BROADCAST_SCORE,
	UNSET_PLAYER_ADMIN,
	RESET_STATES,

	// Opponents actions
	UPDATE_PLAYERS,
	CLEAR_OPPONENTS,
	// Error actions
	SET_ERROR, CLEAR_ERROR,

	// game actions
	SHIFT_TETRO, MORE_TETROS,
	SET_GAME_MODE, SET_GAME_OVER,
	SET_GAME_STARTED, SET_IN_GAME,
	SET_WINNER, RESET_GAME,
	UPDATE_GAME_MODE,
	LEAVE_GAME,
	SET_GAME_TITLE
}
