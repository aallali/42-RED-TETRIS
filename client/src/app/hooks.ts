import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from './store';
import { useGameStatus } from "./TetrisHooks/useGameStatus"
import { useInterval } from "./TetrisHooks/useInterval"
import { usePlayer } from "./TetrisHooks/usePlayer"
import { useStage } from "./TetrisHooks/useStage"


import io from "socket.io-client";

export const socket = io("http://localhost:4242");
export { useGameStatus, usePlayer, useInterval, useStage }
// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
