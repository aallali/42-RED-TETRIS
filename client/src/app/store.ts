import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import playerReducer from "../reducers/player.reducer";
import stageReducer from "../reducers/stage.reducer"
export const store = configureStore({
  reducer: {
    player: playerReducer,
    stage: stageReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
