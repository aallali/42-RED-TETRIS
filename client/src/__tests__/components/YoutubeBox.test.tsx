import renderer from "react-test-renderer";
import YoutubeEmbed from "../../components/YoutubeBox"
import { Provider } from 'react-redux';
import { store } from '../../app/store';

test("_ YouTube Embeded Video Component Test ", () => {
	const tree = renderer.create(<Provider store={store}>
		<YoutubeEmbed embedId="s-Dq5FJEH10" />
	</Provider>).toJSON();
	expect(tree).toMatchSnapshot();
});

