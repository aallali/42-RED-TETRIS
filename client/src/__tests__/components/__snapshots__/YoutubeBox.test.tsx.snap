// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_ YouTube Embeded Video Component Test  1`] = `
<div
  className="video-responsive"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
    height="480"
    src="https://www.youtube.com/embed/s-Dq5FJEH10"
    title="Embedded youtube"
    width="853"
  />
</div>
`;
